test/cbmc/proofs/processDataHandler/processDataHandler_harness.c

```diff
@@ -0,0 +1,134 @@
+/**
+ * @file processDataHandler_harness.c
+ * @brief Implements the proof harness for processDataHandler function.
+ */
+/* Include headers for ota agent. */
+#include "ota.h"
+#include "ota_interface_private.h"
+#include <stdlib.h>
+
+extern OtaAgentContext_t otaAgent;
+extern OtaControlInterface_t otaControlInterface;
+extern OtaErr_t processDataHandler( const OtaEventData_t * pEventData );
+
+/* Stub for ingestDataBlock. */
+IngestResult_t ingestDataBlock( OtaFileContext_t * pFileContext,
+                                const uint8_t * pRawMsg,
+                                uint32_t messageSize,
+                                OtaPalStatus_t * pCloseResult )
+{
+    IngestResult_t result;
+    OtaPalStatus_t status;
+
+    __CPROVER_assert( pFileContext != NULL, "Error: pFileContext cannot be NULL" );
+    __CPROVER_assert( pCloseResult != NULL, "Error: pCloseResult cannot be NULL" );
+    __CPROVER_assert( pRawMsg != NULL, "Error: pRawMsg buffer cannot be NULL." );
+
+    return result;
+}
+
+/* Stub to update the job status. */
+OtaErr_t updateStatus( OtaAgentContext_t * pAgentCtx,
+                       OtaJobStatus_t status,
+                       int32_t reason,
+                       int32_t subReason )
+{
+    OtaErr_t err;
+
+    __CPROVER_assert( pAgentCtx != NULL,
+                      "Error: Ota Agent has to be defined before calling updateJobStatus function." );
+
+    return err;
+}
+
+/* Stub for callback function */
+void otaAppCallback( OtaJobEvent_t eEvent,
+                     const void * pData )
+{
+}
+
+/* Stub for platform specific interface. */
+OtaPalStatus_t setImageState( OtaFileContext_t * const pFileContext,
+                              OtaImageState_t eState )
+{
+    OtaPalStatus_t status;
+
+    __CPROVER_assert( pFileContext != NULL,
+                      "Error: pFileContext is statically defined in otaAgent struct and cannot be NULL." );
+
+    return status;
+}
+
+/* Stub to start the timer event. */
+OtaOsStatus_t start( OtaTimerId_t otaTimerId,
+                     const char * const pTimerName,
+                     const uint32_t timeout,
+                     OtaTimerCallback_t callback )
+{
+    OtaOsStatus_t status;
+
+    __CPROVER_assert( ( otaTimerId != OtaRequestTimer ) || ( otaTimerId != OtaSelfTestTimer ),
+                      "Error: Invalid otaTimerId type. Expected OtaRequestTimer or OtaSelfTestTimer" );
+
+    return status;
+}
+
+void processDataHandler_harness()
+{
+    OtaErr_t err;
+    OtaEventData_t * pEventData;
+    OtaInterfaces_t otaInterface;
+    size_t idx;
+
+    pEventData = ( OtaEventData_t * ) malloc( sizeof( OtaEventData_t ) );
+
+    /* Havoc otaAgent. */
+    __CPROVER_havoc_object( &otaAgent );
+
+    /* The fileType and otaPacketsProcessed cannot exceed the INT32_MAX
+     * and UINT32_MAX respectively. */
+    __CPROVER_assume( otaAgent.fileContext.fileType <= INT32_MAX );
+    __CPROVER_assume( otaAgent.statistics.otaPacketsProcessed < UINT32_MAX );
```

Comment:
Why less than instead of less than or equal?

Reply:
`otaAgent.statistics.otaPacketsProcessed` is incremented in the `processDataHandler` . If it is equal to `UINT32_MAX` then the increment would result in an arithmetic overflow.